Jenkinsfile

```diff
@@ -1,81 +1,108 @@
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.3.0',
+      description: 'The XTC tools version'
+    )
+    string(
+      name: 'XMOSDOC_VERSION',
+      defaultValue: 'v6.0.0',
+      description: 'The xmosdoc version')
+  }
   stages {
-    stage('Standard build and XS3 tests') {
+    stage('Build and test') {
       agent {
-        label 'x86_64&&macOS'
-      }
-      environment {
-        REPO = 'lib_i2c'
-        VIEW = getViewName(REPO)
-      }
-      options {
-        skipDefaultCheckout()
+        label 'x86_64 && linux'
       }
       stages {
-        stage('Get view') {
-          steps {
-            xcorePrepareSandbox("${VIEW}", "${REPO}")
-          }
-        }
-        stage('Library checks') {
-          steps {
-             xcoreLibraryChecks("${REPO}")
-          }
-        }
-        stage('xCORE App XS2 builds') {
+        stage('Build examples') {
           steps {
-            forAllMatch("${REPO}/examples", "app_*/") { path ->
-              runXmake(path)
-            }
-            forAllMatch("${REPO}/examples", "AN*/") { path ->
-              runXmake(path)
+            println "Stage running on ${env.NODE_NAME}"
+
+            dir("${REPO}") {
+              checkout scm
+
+              dir("examples") {
+                withTools(params.TOOLS_VERSION) {
+                  sh 'cmake -G "Unix Makefiles" -B build'
+                  sh 'xmake -C build -j 8'
+                }
+              }
             }
+            runLibraryChecks("${WORKSPACE}/${REPO}")
           }
-        }
-        stage('xCORE App XCOREAI builds') {
+        }  // Build examples
+
+        stage('Build documentation') {
           steps {
-            forAllMatch("${REPO}/examples", "app_*/") { path ->
-              runXmake(path, '', 'XCOREAI=1')
-            }
-            forAllMatch("${REPO}/examples", "AN*/") { path ->
-              runXmake(path, '', 'XCOREAI=1')
+            dir("${REPO}") {
+              withXdoc("v2.0.20.2.post0") {
+                withTools(params.TOOLS_VERSION) {
+                  dir("doc") {
+                    sh "xdoc xmospdf"
+                    archiveArtifacts artifacts: "pdf/*.pdf"
+                  }
+                  forAllMatch("examples", "AN*/") { path ->
+                    dir("${path}/doc")
+                    {
+                      sh "xdoc xmospdf"
+                      archiveArtifacts artifacts: "pdf/*.pdf"
+                    }
+                  }
+                }
+              }
             }
           }
-        }
-        stage('Doc builds') {
+        }  // Build documentation
+
+        stage('Simulator tests') {
           steps {
-            runXdoc("${REPO}/${REPO}/doc")
-            forAllMatch("${REPO}/examples", "AN*/") { path ->
-              runXdoc("${path}/doc")
+            dir("${REPO}") {
+              withTools(params.TOOLS_VERSION) {
+                clone_test_deps()
+                createVenv(reqFile: "tests/requirements.txt")
```

Comment:
As it's written, the relative path on the pip install of requirements.txt assumes you are in `lib_i2c` rather than `lib_i2c/tests`. I think it would be more natural for someone to run pip install from inside the tests directory. So I think this would be better as:
```
dir("tests") {
  createVenv(reqFile: "requirements.txt")
```
and then requirements.txt needs `-e ./../../test_support`

Reply:
That makes sense. I'll change. Thanks

Reply:
piping in here cause this came up in my notifications to say - I always create the .venv for the component I'm working in the root directory of the library, so e.g. `~/sandboxes/i2c_work_this_sprint/lib_i2c` I'll instantiate the .venv there, which is I think in contrast to the statement here - but that's a style thing, right?

There's no README-DEV in the top level or readme in the tests so I guess there's nowhere to communicate this - so I think I'd suggest that it be "however the other lib_ repos do it and we just make that culturally known"... however that is...